src/merkle/tiered_smt/proof.rs

```diff
@@ -110,6 +111,20 @@ impl TieredSmtProof {
             .expect("failed to compute Merkle path root")
     }
 
+    /// Consume the proof and returns its parts.
+    pub fn into_parts(self) -> (MerklePath, Vec<(RpoDigest, Word)>) {
+        (self.path, self.entries)
+    }
+
+    /// Create a proof from its parts
+    pub fn from_parts<I>(path: MerklePath, entries: I) -> Self
+    where
+        I: IntoIterator<Item = (RpoDigest, Word)>,
+    {
+        let entries: Vec<(RpoDigest, Word)> = entries.into_iter().collect();
+        Self { path, entries }
```

Comment:
If we chose this pattern then maybe we should invoke the constructor so that we apply the assertions it contains? `Self::new(path, entries)`